### ISSUE TITLE
Calculate audited distance for users w/ audit_task table instead of mission table

### ISSUE THREAD

**Issue by misaugstad** [ProjectSidewalk/SidewalkWebpage#3085]
##### Brief description of problem/feature
TLDR; we should use the `audit_task` table instead of the `mission` table to calculate the distance that user's have audited, because it will be more precise on average, and it leans more towards being lenient when determining whether to mark someone as low quality (i.e., it's more likely to underestimate audited distance, and therefore overestimate labeling frequency).

---

We had some recent bugs on the Explore page that caused users to see "mission complete" messages way too frequently, often when loading the page for the first time. These should be fixed as of PR #2979 

But one side effect is that for the users that were auditing during that time frame where the bug existed, they have some completed missions in the database that shouldn't be there, and are nearly impossible to find and remove. I think that this may have only been an issue on test servers, but it definitely impacted our new Zurich test server!

As such, calculating how much a user has audited by using the `mission` table is less accurate now. And it always had more variance due to various bugs over the years on the Explore page. Because of this, I think that we should move the remaining queries that calculate distance using the `mission` table to using the `audit_task` table.

In Zurich, this has marked some users as "low quality" when they shouldn't be! We have one user that has been testing out Project Sidewalk there, and they happened to be testing during the two or so weeks when the bug was live. The `mission` table says that they've audited 3,150 meters, whereas the `audit_task` table says that they've audited about 1,939 meters. This was enough to get them incorrectly marked as "low quality"!

##### Potential solution(s)
The `mission` table has a `distance_progress` column (in meters), and we were just summing that across all of a user's missions. Now what we have to do (and we already do this in many other places) is to use postgis to measure the length of the streets that have been completed by the user, which are marked in the `audit_task` table.

This does under count the distance that has been audited when a user has streets that they only partially completed. It's a little bit more work, but if it matters, we should be able to calculate how much of a street a user audited if they didn't finish it. The `audit_task` table has `current_lat` and `current_lng` columns, which will tell us how far they got on the street. The downsides to using this are that it's some more code complexity, I'm not 100% sure that we have access to all the functions that we need in our `slick` library (so we might have to convert to raw SQL), and those lat/lngs have been known to not always be correct, again due to front-end bugs.

My vote is to plan to include data on partially audited streets. Then if I discover that it's more complicated than I was hoping while working on implementation, I can scale it back to just look at completed streets.
